OpenRA.Mods.Common/Scripting/Global/ActorGlobal.cs

```diff
@@ -71,7 +71,19 @@ public int BuildTime(string type)
 			if (!Context.World.Map.Rules.Actors.TryGetValue(type, out ai))
 				throw new LuaException("Unknown actor type '{0}'".F(type));
 
-			return ai.GetBuildTime();
+			var bi = ai.TraitInfoOrDefault<BuildableInfo>();
+
+			if (bi != null)
+			{
+				var time = bi.BuildDuration != -1
+					? (ai.HasTraitInfo<ValuedInfo>() ? ai.TraitInfo<ValuedInfo>().Cost : 0)
+					: bi.BuildDuration;
+
+				time *= bi.BuildDurationMultiplier / 100;
```

Comment:
This is integer division, so you need to do it *after* the multiplication: `time = time * bi.BuildDurationMultiplier / 100`.